the-collab-lab/tcl-13-smart-shopping-list
# At rb welcome message

## Description

If the user does not have any items in their list, they are presented with a prompt and link to Add Items to add their first items to their list. ListContextProvider was updated so that `[userList, setUserList]` had an empty array in `useState` so that the `.length` method could be called in ViewList. This allowed a prompt for users to add an item to their list if the list was empty. If the list is not empty, the database is accessed and the list items are shown. 
 
Refactored App.js so that the nav doesn't render unless there's a token in `localStorage`, because when there is no token in `localStorage` the Add Items and View List links will not work and the user is redirected to Welcome. Router was also refactored to redirect to Welcome if there is no user token in `localStorage`. 


## Related Issue

Closes #7 

## Acceptance Criteria


## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
| ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

**ViewList**
Refactored the conditional rendering of the list items. Instead of returning null, if there are no items the user is shown a prompt telling them that they have no items and a link to Add Items. Uses `.length` on `currentList.userList` to check if there's any items in `userList`. 

**ListContext**
Added an empty array to `useState` from `[userList, setUserList]` so that the `.length` method can be called. 

**App.js** 
Refactored the Nav so that it only renders when there is a user token saved to `localStorage`.

**Router**
Redirects users to the Welcome component if there is no token in `localStorage`. 

### Before

![Screen Shot 2020-10-26 at 12 47 45 PM](https://user-images.githubusercontent.com/49924080/97202091-8e486680-1789-11eb-8528-ae3ab2116a02.png)

### After

Nav Links do not appear if there is no token in `localStorage`
![Screen Shot 2020-10-26 at 12 51 07 PM](https://user-images.githubusercontent.com/49924080/97202619-3a8a4d00-178a-11eb-968e-0f7d37d09ce3.png)

User prompt and link to add items
![Screen Shot 2020-10-26 at 12 51 22 PM](https://user-images.githubusercontent.com/49924080/97202660-48d86900-178a-11eb-824e-b28b64266615.png)

## Testing Steps / QA Criteria

Clear any existing tokens from `localStorage`. On app load, you should be redirected to the Welcome component. The View List and Add Items links (Nav links) should not appear at the bottom of the screen. After creating a new list, you should be redirected to View List, the Nav links should appear, and a prompt should inform you that your list is empty and show a link to add items.
